plugins/inputs/varnish/varnish_test.go

```diff
@@ -427,3 +429,212 @@ LCK.pipestat.creat                                       1         0.00 Created
+			activeVcl:   "VCL_xxxx_xxx_VOD_SHIELD_Vxxxxxxxxxxxxx_xxxxxxxxxxxxx",
+			field:       "bereq_hdrbytes",
+		},
+		{
+			vName:       "VBE.VCL_ROUTER_V123_123.default.happy",
+			tags:        map[string]string{"backend": "default"},
+			field:       "happy",
+			activeVcl:   "VCL_ROUTER_V123_123",
+			measurement: "varnish_vbe"},
+		{
+			vName:       "KVSTORE.ds_stats.VCL_xxxx_xxx_A_B_C.shield",
+			tags:        map[string]string{"id": "ds_stats"},
+			field:       "shield",
+			activeVcl:   "VCL_xxxx_xxx_A_B_C",
+			measurement: "varnish_kvstore",
+		},
+		{
+			vName:       "LCK.goto.director.destroy",
+			tags:        map[string]string{"id": "goto.director"},
+			field:       "destroy",
+			activeVcl:   "",
+			measurement: "varnish_lck",
+		},
+		{
+			vName:       "XCNT.1111.XXX+_LINE.cr.deliver_stub_restart.val",
+			tags:        map[string]string{"group": "XXX+_LINE.cr"},
+			field:       "deliver_stub_restart",
+			activeVcl:   "1111",
+			measurement: "varnish_xcnt",
+		},
+		{
+			vName:       "XCNT.vcl123.my_field.val",
+			tags:        map[string]string{},
+			field:       "my_field",
+			activeVcl:   "vcl123",
+			measurement: "varnish_xcnt",
+		},
+	} {
+		vMetric := parseMetricV2(c.vName)
+		require.Equal(t, c.activeVcl, vMetric.vclName)
+		require.Equal(t, c.measurement, vMetric.measurement, c.vName)
+		require.Equal(t, c.field, vMetric.fieldName)
+		require.Equal(t, c.tags, vMetric.tags)
+	}
+}
+
+func TestVersions(t *testing.T) {
+	server := &Varnish{}
+	require.Equal(t, "A plugin to collect stats from Varnish HTTP Cache", server.Description())
+	acc := &testutil.Accumulator{}
+
+	require.Equal(t, 0, len(acc.Metrics))
+
+	type testConfig struct {
+		jsonFile           string
+		activeReloadPrefix string
+		size               int
+	}
+
+	for _, c := range []testConfig{
+		{jsonFile: "varnish_types.json", activeReloadPrefix: "", size: 3},
+		{jsonFile: "varnish6.2.1_reload.json", activeReloadPrefix: "reload_20210623_170621_31083", size: 374},
+		{jsonFile: "varnish6.2.1_reload.json", activeReloadPrefix: "", size: 434},
+		{jsonFile: "varnish6.6.json", activeReloadPrefix: "boot", size: 358},
+		{jsonFile: "varnish4_4.json", activeReloadPrefix: "boot", size: 295},
+	} {
+		output, _ := ioutil.ReadFile("test_data/" + c.jsonFile)
+		err := server.processMetricsV2(c.activeReloadPrefix, acc, bytes.NewBuffer(output))
+		require.NoError(t, err)
+		require.Equal(t, c.size, len(acc.Metrics))
+		for _, m := range acc.Metrics {
+			require.NotEmpty(t, m.Fields)
+			require.Contains(t, m.Measurement, "varnish_")
+			for field := range m.Fields {
+				require.NotContains(t, field, "reload_")
+			}
+			for tag := range m.Tags {
+				require.NotContains(t, tag, "reload_")
+			}
+		}
+		acc.ClearMetrics()
+	}
+}
+
+func TestJsonTypes(t *testing.T) {
+	json := `{
+		"timestamp": "2021-06-23T17:06:37",
+			"counters": {
+			"XXX.floatTest": {
+				"description": "floatTest",
+					"flag": "c",
```

Comment:
Are these flags and formats something that is provided from varnish metrics?

Reply:
Flags and formats are used to display counter in varnishstat interactive text UI. `flag="b"` is used only for `happy` counter, which contains health probes encoded into btimap. Happy probes are bits set to 1, and the unhappy ones are set to 0. The highest bits represent the oldest probes.